lib/WorseReflection/Bridge/TolerantParser/Diagnostics/MissingReturnTypeProvider.php

```diff
@@ -144,11 +162,16 @@ public function exit(NodeContextResolver $resolver, Frame $frame, Node $node): i
             return;
         }
 
-        if ($method->docblock()->returnType()->isMixed()) {
+        $docblockReturn = $method->docblock()->returnType();
+        if ($docblockReturn->isMixed()) {
+            return;
+        }
+
+        if ($docblockReturn->isDefined()) {
```

Comment:
There are `MissingReturnTypeProvider` and `DocblockMissingReturnTypeProvider`. It seems, even on `master` branch declaring return type only in docblock leaves with false-positive diagnostic about missing return type. I've added this check, but it may be better to merge both classes -- despite the change I think it is out of the scope of this PR, but without this new test case is failing. Anyway, up to you if this should be split into own PR.